src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -1083,131 +997,140 @@ public boolean remove(Runnable task) {
-        ClockWrapper.stopForcingUpdate();
-        try {
-          int index = ListUtils.getInsertionEndIndex(scheduleQueue, task, true);
-          
-          scheduleQueue.add(index, task);
-        } finally {
-          ClockWrapper.resumeForcingUpdate();
-        }
+        clearQueue(scheduleQueue, removedTasks);
       }
-      
-      handleQueueUpdate();
     }
-
-    /**
-     * Adds a scheduled task to the end of the scheduled queue.  It is expected that this task is 
-     * NOT ready for execution, and will later be moved from invoking 
-     * {@link #reschedule(RecurringTaskWrapper)}.
-     * 
-     * @param task Task to add to end of schedule queue
-     */
-    public void addScheduledLast(RecurringTaskWrapper task) {
-      scheduleQueue.addLast(task);
-      // no need to notify since this task wont be ready to run
+  
+    private static void clearQueue(Collection<? extends TaskWrapper> queue, List<TaskWrapper> resultList) {
+      Iterator<? extends TaskWrapper> it = queue.iterator();
+      while (it.hasNext()) {
+        TaskWrapper tw = it.next();
+        tw.cancel();
+        if (! (tw.task instanceof ShutdownRunnable)) {
+          int index = ListUtils.getInsertionEndIndex(resultList, tw, true);
+          resultList.add(index, tw);
+        }
+      }
+      queue.clear();
     }
-
-    /**
-     * Call to find and reposition a scheduled task.  It is expected that the task provided has 
-     * already been added to the queue (likely from a call to 
-     * {@link #addScheduledLast(RecurringTaskWrapper)}).  This call will use 
-     * {@link RecurringTaskWrapper#getNextDelayInMillis()} to figure out what the new position 
-     * within the queue should be.
-     * 
-     * @param task Task to find in queue and reposition based off next delay
-     */
-    public void reschedule(RecurringTaskWrapper task) {
-      synchronized (scheduleQueue.getModificationLock()) {
-        if (! workerPool.isShutdownStarted()) {
+    
+    public TaskWrapper getNextTask() {
+      TaskWrapper nextScheduledTask = scheduleQueue.peekFirst();
+      TaskWrapper nextExecuteTask = executeQueue.peek();
+      if (nextExecuteTask != null) {
+        if (nextScheduledTask != null) {
+          long scheduleDelay;
+          long executeDelay;
           ClockWrapper.stopForcingUpdate();
           try {
-            long nextDelay = task.getNextDelayInMillis();
-            int insertionIndex = ListUtils.getInsertionEndIndex(scheduleQueue, nextDelay, true);
-            
-            scheduleQueue.reposition(task, insertionIndex, true);
+            scheduleDelay = nextScheduledTask.getDelay(TimeUnit.MILLISECONDS);
+            executeDelay = nextExecuteTask.getDelay(TimeUnit.MILLISECONDS);
           } finally {
             ClockWrapper.resumeForcingUpdate();
           }
+          if (scheduleDelay < executeDelay) {
+            return nextScheduledTask;
+          } else {
+            return nextExecuteTask;
```

Comment:
Yes, we have to pick one to favor.  In general it should not make any sense because executed or scheduled tasks should not require accurate time execution.